### PR TITLE
Init vikunja package and module

### PR DESCRIPTION
**Pull request by leona-ya**
###### Motivation for this change
Add support for the [vikunja to-do list app](https://vikunja.io).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126531` at c95e38f8 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>vikunja-api</li>
    <li>vikunja-frontend</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -2983,6 +2983,12 @@
     githubId = 57923898;
     name = "Elyhaka";
   };
+  em0lar = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:2986]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/servers/web-apps/vikunja/api.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{ lib, buildGoModule, fetchgit }:
+
+buildGoModule rec {
+  pname = "vikunja-api";
+  version = "0.17.0";
+
+  src = fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/servers/web-apps/vikunja/api.nix:7]:
```suggestion
  src = fetchFromGitea {
```

File: pkgs/servers/web-apps/vikunja/frontend.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{ stdenv, lib, fetchurl, unzip, ... }:
+
+stdenv.mkDerivation rec {
+  pname = "vikunja-frontend";
+  version = "0.17.0";
+  src = fetchurl {
+    url = "https://dl.vikunja.io/frontend/${pname}-${version}.zip";
```
Comment (SuperSandro2000) [inline pkgs/servers/web-apps/vikunja/frontend.nix:7]: Can we compile this ourselves?

File: pkgs/servers/web-apps/vikunja/frontend.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  nativeBuildInputs = [ unzip ];
+
+  unpackPhase = ''
+    runHook preUnpack
+
+    unzip $src -d vikunja-frontend
+
+    runHook postUnpack
+  '';
```
Comment (SuperSandro2000) [inline pkgs/servers/web-apps/vikunja/frontend.nix:19]:
```suggestion
  sourceRott = ".";
```

**Review by aanderse — CHANGES_REQUESTED** [review]
Hi! Thanks for contributing a module to nixpkgs! It looks like a solid start. I have left some suggestions on how we might improve things.

I don't know how experienced you are with nix and/or writing NixOS modules so I'll state that I'm happy to answer any questions, review any ideas, or even lend a helping hand on this PR *if* you need it - don't hesitate to ask.

File: nixos/modules/services/web-apps/vikunja.nix [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+          ensurePermissions = { "DATABASE ${cfg.config.database.database}" = "ALL PRIVILEGES"; };
+        }
+      ];
+    };
+
+    services.mysql = optionalAttrs (useMysql && cfg.config.database.createDatabase) {
+      enable = mkDefault true;
+      package = mkDefault pkgs.mariadb;
+
+      ensureDatabases = [ cfg.config.database.database ];
```
[the remark this answers is not in the document]
Comment (em0lar) [reply nixos/modules/services/web-apps/vikunja.nix]: I have a problem with that. I don't have much experience with mariadb. I could try to build something that sets the database charsets and collations correctly, but I won't be sure it's correct. Only postgresql I am sure about. What do you think: should I try or should I leave the creation of the database (maybe only when the user chooses mariadb) to the user? (even if this would not be a complete nixos module).

File: nixos/modules/services/web-apps/vikunja.nix [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+          tryFiles = "try_files $uri $uri/ /";
+        };
+        "~* ^/(api|dav|\.well-known)/" = {
+          proxyPass = "http://localhost:3456";
+          extraConfig = ''
+            client_max_body_size 20M;
+          '';
+        };
+      };
+    };
```
[the remark this answers is not in the document]
Comment (em0lar) [reply nixos/modules/services/web-apps/vikunja.nix]: At what point do you want an explanation or an option? For me it would also be ok to serve over SSL by default, but I don't think everyone wants that.

For example, the nextcloud nixos module doesn't have SSL enabled by default either. There you have to activate it yourself, for example with:
```
services.nginx.virtualHosts."${config.services.nextcloud.hostName}" = {
  enableACME = true;
  forceSSL = true;
};
```

File: nixos/modules/services/web-apps/vikunja.nix [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+    frontendHostname = mkOption {
+      type = types.str;
+      description = "The Hostname under which the frontend is running.";
+    };
+    extraConfig = mkOption {
+      type = types.attrs;
+      default = {};
+      description = "Extra config that fill directly be passed into the Vikunja config.";
+    };
+    config = {
```
[the remark this answers is not in the document]
Comment (em0lar) [reply nixos/modules/services/web-apps/vikunja.nix]: Thanks for the idea with the configuration options. I took a bit of time to build a NixOS test, but since it's my first NixOS test, I'm really not sure if I did it well, so feel free to suggest improvements,

File: pkgs/servers/web-apps/vikunja/api.nix [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    repo = "api";
+    rev = "v${version}";
+    sha256 = "sha256-A9ZrJlGT2QZ9qPz/9XPgsBJqv1PA9Kb5scnXWzZnosc=";
+  };
+
+  vendorSha256 = "sha256-/vXyZznGxj5hxwqi4sttBBkEoS25DJqwoBtADCRO9Qc=";
+
+  # checks need to be disabled because of needed internet for some checks
+  doCheck = false;
+
```
Comment (jtojnar) [inline pkgs/servers/web-apps/vikunja/api.nix:19]: According to [build instructions](https://vikunja.io/docs/build-from-sources/), it should be built with Mage ([example](https://github.com/jtojnar/nixfiles/blob/28a348c486e749f15be23f59e7cbea8efb08378f/pkgs/vikunja/vikunja-api/default.nix)). I am surprised this even builds – in the past, I had to [build templates](https://kolaente.dev/vikunja/api/issues/134).

File: pkgs/servers/web-apps/vikunja/frontend.nix (hunk 3 above)
Comment (jtojnar) [reply pkgs/servers/web-apps/vikunja/frontend.nix:7]: It is certainly possible with yarn2nix ([example](https://github.com/jtojnar/nixfiles/blob/28a348c486e749f15be23f59e7cbea8efb08378f/pkgs/vikunja/vikunja-frontend/default.nix)). But there is a bug: https://github.com/nix-community/yarn2nix/issues/173

File: nixos/tests/vikunja.nix [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+        database = {
+          type = "postgres";
+          user = "vikunja";
+          database = "vikunjadb";
+        };
+        frontendScheme = "http";
+        frontendHostname = "localhost";
+        environmentFiles = [
+          (pkgs.writeText "vikunja-env" ''
+            VIKUNJA_DATABASE_PASSWORD=snakeoilpwd
```
Comment (jtojnar) [inline nixos/tests/vikunja.nix:32]: Might be nicer to go passwordless and connect through Unix sockets: https://github.com/jtojnar/nixfiles/blob/28a348c486e749f15be23f59e7cbea8efb08378f/hosts/azazel/ogion.cz/todo/default.nix#L59

File: nixos/modules/services/web-apps/vikunja.nix [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+      };
+    };
+
+    services.nginx.virtualHosts."${cfg.frontendHostname}" = mkIf cfg.setupNginx {
+      locations = {
+        "/" = {
+          root = cfg.package-frontend;
+          tryFiles = "try_files $uri $uri/ /";
+        };
+        "~* ^/(api|dav|\.well-known)/" = {
```
Comment (jtojnar) [inline nixos/modules/services/web-apps/vikunja.nix:133]: The `\` needs to be escaped – Nix interprets `\.` as `.`.

File: nixos/tests/vikunja.nix (hunk 9 above)
Comment (em0lar) [reply nixos/tests/vikunja.nix:32]: No, that wouldn't be nicer. The reason for that is that the module creates a systemd service with a dynamic user. As far as I know postgresql peer authentication is not possible with dynamic users.

File: nixos/doc/manual/release-notes/rl-2111.section.md [hunk 11]
```diff
@@ -16,6 +16,9 @@ In addition to numerous new and upgraded packages, this release has the followin
 
 - [ucarp](https://download.pureftpd.org/pub/ucarp/README), an userspace implementation of the Common Address Redundancy Protocol (CARP). Available as [networking.ucarp](options.html#opt-networking.ucarp.enable).
 
+- [vikunja](https://vikunja.io), an to-do list app. Available as
+  [services.vikunja](options.html#opt-services.vikunja.enable).
```
[the remark this answers is not in the document]
Comment (em0lar) [reply nixos/doc/manual/release-notes/rl-2111.section.md:20]: Are you sure? It works for me when I build the docs.

**Comment by jtojnar** [comment]
It works but it won't check for typos if you use anything other than an
anchor.

File: nixos/doc/manual/release-notes/rl-2111.section.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (em0lar) [reply nixos/doc/manual/release-notes/rl-2111.section.md:20]: > It works but it won't check for typos if you use anything other than an
anchor.

Then I don't understand why the other people used it in this file.

**Comment by jtojnar** [comment]
Likely because they are not familiar with our docs system so they felt back
to using relative URLs.

File: nixos/tests/vikunja.nix (hunk 9 above)
Comment (aanderse) [reply nixos/tests/vikunja.nix:32]: I never got around to testing if `DynamicUser` works with socket authentication on `postgresql` but it **does** with `mariadb`. You tested and confirms it does not?

File: nixos/modules/services/web-apps/vikunja.nix (hunk 5 above)
[the remark this answers is not in the document]
Comment (aanderse) [reply nixos/modules/services/web-apps/vikunja.nix]: It a basic test works and you don't have any reason to believe the encoding is wrong you could leave it as is. Maybe that is easiest for now.

[the remark this answers is not in the document]
Comment (em0lar) [reply nixos/modules/services/web-apps/vikunja.nix]: For me it would also be okay to let the user create the database

File: nixos/modules/services/web-apps/vikunja.nix (hunk 6 above)
[the remark this answers is not in the document]
Comment (aanderse) [reply nixos/modules/services/web-apps/vikunja.nix]: I think a little documentation would go a long way. The `nextcloud` module has an entire section in the manual. Maybe just a few lines that inform the sysadmin they are responsible for configuring `nginx` and a practical example of what most users should do

File: nixos/modules/services/web-apps/vikunja.nix (hunk 5 above)
[the remark this answers is not in the document]
Comment (aanderse) [reply nixos/modules/services/web-apps/vikunja.nix]: That sounds good :+1:

File: nixos/modules/services/web-apps/vikunja.nix (hunk 6 above)
[the remark this answers is not in the document]
Comment (em0lar) [reply nixos/modules/services/web-apps/vikunja.nix]: Do you have a recommendation where I should add the documentation?

[the remark this answers is not in the document]
Comment (aanderse) [reply nixos/modules/services/web-apps/vikunja.nix]: I think adding it to the `description` of `setupNginx` is probably a good spot. Keep it short and simple and I think it will provide a large amount of value for a small amount of effort :+1:

File: pkgs/servers/web-apps/vikunja/frontend.nix (hunk 3 above)
Comment (em0lar) [reply pkgs/servers/web-apps/vikunja/frontend.nix]: I pushed a version based on @jtojnar's example. Maybe you can have a look at it and check if it's good.

File: nixos/modules/services/web-apps/vikunja.nix (hunk 6 above)
[the remark this answers is not in the document]
Comment (em0lar) [reply nixos/modules/services/web-apps/vikunja.nix]: Am I understanding correctly that you want me to configure nginx without SSL in the module and add some lines on how to enable SSL in the `setupNginx` `description`?

[the remark this answers is not in the document]
Comment (aanderse) [reply nixos/modules/services/web-apps/vikunja.nix]: Sure, that sounds good to me.

File: pkgs/servers/web-apps/vikunja/frontend.nix (hunk 3 above)
Comment (kolaente) [reply pkgs/servers/web-apps/vikunja/frontend.nix]: My attempt at packaging is [here](https://kolaente.dev/vikunja/desktop-nix/src/branch/main/frontend/default.nix) but it has the wired error from the bug @jtojnar mentioned. I did have some success building it on 21.05 but not reliably.

File: nixos/modules/services/web-apps/vikunja.nix [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+    systemd.services.vikunja-api = {
+      description = "vikunja-api";
+      after = [ "network.target" ] ++ lib.optional usePostgresql "postgresql.service" ++ lib.optional useMysql "mysql.service";
+      wantedBy = [ "multi-user.target" ];
+      path = [ cfg.package-api ];
+      restartTriggers = [ configFile ];
+
+      serviceConfig = {
+        Type = "simple";
+        DynamicUser = true;
```
Comment (kolaente) [inline nixos/modules/services/web-apps/vikunja.nix:119]: I'm not sure if I'm understanding systemd's DynamicUser's well enough, but does this mean the service is executed with a different uid every time it starts? Because if that's the case that would break Vikunja's access to uploaded files.

File: nixos/modules/services/web-apps/vikunja.nix [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+        "~* ^/(api|dav|\\.well-known)/" = {
+          proxyPass = "http://localhost:3456";
+          extraConfig = ''
+            client_max_body_size 20M;
+          '';
+        };
+      };
+    };
+
+    environment.etc."vikunja/config.json".source = configFile;
```
Comment (kolaente) [inline nixos/modules/services/web-apps/vikunja.nix:142]: Is this regenerating the `yaml` configuration generated in the `configFile` variable but this time as a json config?

File: nixos/tests/vikunja.nix [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+
+  testScript =
+    ''
+      vikunjaSqlite.wait_for_unit("vikunja-api.service")
+      vikunjaSqlite.wait_for_open_port(3456)
+      vikunjaSqlite.succeed("curl --fail http://localhost:3456/api/v1/info")
+
+      vikunjaSqlite.wait_for_unit("nginx.service")
+      vikunjaSqlite.wait_for_open_port(80)
+      vikunjaSqlite.succeed("curl --fail http://localhost/api/v1/info")
```
Comment (kolaente) [inline nixos/tests/vikunja.nix:55]: Might want to add another check for the frontend as well to see if it is served correctly, not only if the nginx proxies the api requests correctly.

If we want to go really fancy we could also compare the outputs of `http://localhost:3456/api/v1/info` and `http://localhost/api/v1/info` but I'm not sure if that has any benefit.

File: pkgs/servers/web-apps/vikunja/api.nix (hunk 8 above)
Comment (kolaente) [reply pkgs/servers/web-apps/vikunja/api.nix:19]: We got rid of the template build step since a while ago.

Not building with mage won't set the version and may fail with sqlite as that needs cgo to work correctly (I guess the test would catch that though) so I think while building with a plain `go build .` (I'm assuming `buildGoModule` does only that?) will probably work, it would be better to switch the build to mage.

File: pkgs/servers/web-apps/vikunja/frontend.nix [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+    owner = "vikunja";
+    repo = "frontend";
+    rev = "v${version}";
+    sha256 = "sha256-/KCpsPEhCJz+1pLisSviFP/VCiXUD6N9l1ZragOd2Go=";
+  };
+  frontend-modules = mkYarnPackage rec {
+    name = "vikunja-frontend-modules";
+    inherit version src;
+
+    packageJSON = ./frontend-package.json;
```
Comment (kolaente) [inline pkgs/servers/web-apps/vikunja/frontend.nix:17]: Is it really required to keep the `package.json` and generated `yarn2nix` files in the repo? AFAIU `yarn2nix` will regenerate them based on the content of the repo itself so these are rather redundant.

File: pkgs/servers/web-apps/vikunja/frontend.nix [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+  pname = "vikunja-frontend";
+  inherit version src;
+
+  nativeBuildInputs = [ frontend-modules yarn ];
+
+  buildPhase = ''
+    # Cannot use symlink or postcss-loader will crap out
+    cp -r ${frontend-modules}/libexec/vikunja-frontend/node_modules/ .
+    yarn --offline run build
+    # Unfortunately, this needs to be hardcoded at build.
```
Comment (kolaente) [inline pkgs/servers/web-apps/vikunja/frontend.nix:33]: Actually, this is only kind of true: You can configure the api url at runtime in the frontend itself, it will only be pre-filled with the value set like this. If the variable is empty it will show the input without any prompts directly. We're [doing this for the desktop app](https://kolaente.dev/vikunja/desktop/src/branch/main/.drone.yml#L49).

**Review by kolaente — COMMENTED** [review]
Awesome to see this getting packaged for nix!

My nix-packaging knowledge is rather limited, but from the Vikunja side of things this looks good.

File: nixos/modules/services/web-apps/vikunja.nix (hunk 6 above)
[the remark this answers is not in the document]
Comment (em0lar) [reply nixos/modules/services/web-apps/vikunja.nix]: I added something. You could have a look whether that's good

File: nixos/tests/vikunja.nix (hunk 9 above)
Comment (em0lar) [reply nixos/tests/vikunja.nix]: I tested (again) and it seems to work (requirement: database user has to be named the same as systemd service). So I changed the test to passwordless login.

File: nixos/modules/services/web-apps/vikunja.nix (hunk 13 above)
Comment (em0lar) [reply nixos/modules/services/web-apps/vikunja.nix]: Ah, thanks for the indirect hint. In this line is specified that the file in `configFile` is linked to `/etc/vikunja/config.json`. But that should be `/etc/vikunja/config.yaml`. I changed that.

File: nixos/modules/services/web-apps/vikunja.nix (hunk 12 above)
Comment (em0lar) [reply nixos/modules/services/web-apps/vikunja.nix:122]: The first part is correct. The second not really (if I understood it correctly). The files are stored in `/var/lib/private/...` and symlinked to `/var/lib/vikunja` (the `StateDirectory`) with ownership to the dynamic user. So systemd manages that the StateDirectory is owned by the dynamic user. More details over [here](https://noise.getoto.net/2017/10/06/dynamic-users-with-systemd/).

File: pkgs/servers/web-apps/vikunja/frontend.nix (hunk 16 above)
Comment (em0lar) [reply pkgs/servers/web-apps/vikunja/frontend.nix:33]: What do you think I should do? I think that is the best solution (with maybe changing the comment).

File: nixos/doc/manual/release-notes/rl-2111.section.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (em0lar) [reply nixos/doc/manual/release-notes/rl-2111.section.md:20]: I think that you're wrong. This are the release notes, not the (module) options. So I need to link to "options.html#...".

File: nixos/tests/vikunja.nix (hunk 14 above)
Comment (em0lar) [reply nixos/tests/vikunja.nix:53]: I added a check for the frontend part. The other one I would save for the future

File: pkgs/servers/web-apps/vikunja/frontend.nix (hunk 3 above)
Comment (em0lar) [reply pkgs/servers/web-apps/vikunja/frontend.nix]: Do you think it would be better to use the prebuild version at the moment and switch to a self-build version later when it's more stable (I don't know how unstable it is because I didn't experience the issue myself)?

File: nixos/modules/services/web-apps/vikunja.nix (hunk 7 above)
[the remark this answers is not in the document]
Comment (em0lar) [reply nixos/modules/services/web-apps/vikunja.nix]: Should I add `passthru.tests = nixosTests.vikunja` to the vikunja-api or vikunja-frontend package or both? The NixOS test mostly tests the api but also a bit of the frontend.

File: nixos/modules/services/web-apps/vikunja.nix (hunk 6 above)
[the remark this answers is not in the document]
Comment (aanderse) [reply nixos/modules/services/web-apps/vikunja.nix]: Perfect!

File: pkgs/servers/web-apps/vikunja/frontend.nix (hunk 3 above)
Comment (kolaente) [reply pkgs/servers/web-apps/vikunja/frontend.nix]: If it worked for you then we should be fine building this ourselves.

File: nixos/modules/services/web-apps/vikunja.nix (hunk 12 above)
Comment (kolaente) [reply nixos/modules/services/web-apps/vikunja.nix:122]: Okay, that makes a lot of sense then. Thanks for clarifying!

File: pkgs/servers/web-apps/vikunja/frontend.nix (hunk 16 above)
Comment (kolaente) [reply pkgs/servers/web-apps/vikunja/frontend.nix:33]: I think this is fine. It could be changed later on, but for now it is fine.

File: pkgs/servers/web-apps/vikunja/frontend.nix (hunk 3 above)
Comment (em0lar) [reply pkgs/servers/web-apps/vikunja/frontend.nix]: I have just taken some time to test again exactly and then this error also occurred several times. Therefore I would suggest to take the pre-build variant again, if not someone has an idea for a fix.

**Comment by em0lar** [comment]
* Set `services.vikunja.settings.files.basepath` setting to `${cfg.stateDir}/files` by default to fix file management (probably broken due to DynamicUser / user without home directory)
* Append slash to `services.vikunja.settings.service.frontendurl setting` to fix for example OIDC redirects

**Comment by em0lar** [comment]
* reverted vikunja-frontend to pre-build variant (see https://github.com/NixOS/nixpkgs/pull/126531#discussion_r659315549 for exaplanation)
* fix in commit history (vikunja-api was added to /pkgs/top-level/all-packages.nix in vikunja-frontend init commit)

File: pkgs/servers/web-apps/vikunja/frontend.nix (hunk 15 above)
Comment (em0lar) [reply pkgs/servers/web-apps/vikunja/frontend.nix]: No longer important, since reverted to pre-build version.

File: nixos/doc/manual/release-notes/rl-2111.section.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (jtojnar) [reply nixos/doc/manual/release-notes/rl-2111.section.md:20]: The documentation system supports two kinds of links – internal and external. When you use just the hash fragment, it is considered an internal link that will be resolved to an anchor anywhere in the book. If you use file name, it will be considered an external link.

[the remark this answers is not in the document]
Comment (em0lar) [reply nixos/doc/manual/release-notes/rl-2111.section.md]: Okay, thank you and sorry for not changing it before. Now I changed it in fc5dc6d77f4bee73a9b4fcc1a10c0871e240fcff.

File: nixos/modules/services/web-apps/vikunja.nix (hunk 7 above)
[the remark this answers is not in the document]
Comment (em0lar) [reply nixos/modules/services/web-apps/vikunja.nix]: Hey @aanderse could you have a look at this question?

[the remark this answers is not in the document]
Comment (aanderse) [reply nixos/modules/services/web-apps/vikunja.nix]: I'm sorry @em0lar I missed this comment. Thanks for pinging me. Sounds like it makes sense to add to both.

File: pkgs/servers/web-apps/vikunja/frontend.nix (hunk 3 above)
Comment (em0lar) [reply pkgs/servers/web-apps/vikunja/frontend.nix:7]: I decided to use the pre-build version until someone has a fix

File: nixos/modules/services/web-apps/vikunja.nix (hunk 7 above)
[the remark this answers is not in the document]
Comment (em0lar) [reply nixos/modules/services/web-apps/vikunja.nix]: Thank you @aanderse for your reply. I added it to both packages

File: nixos/modules/services/web-apps/vikunja.nix [hunk 17]
```diff
@@ -0,0 +1,150 @@
...
+      defaultText = "pkgs.vikunja-api";
+      description = "vikunja-api derivation to use.";
+    };
+    package-frontend = mkOption {
+      default = pkgs.vikunja-frontend;
+      type = types.package;
+      defaultText = "pkgs.vikunja-frontend";
+      description = "vikunja-frontend derivation to use.";
+    };
+    stateDir = mkOption {
```
Comment (aanderse) [inline nixos/modules/services/web-apps/vikunja.nix:26]: Since you are using `DynamicUser` this directory is constrained to exist under `/var/lib`. I think this value doesn't provide much value and might cause confusion. If I were you I would just hardcode the state directory.

**Review by aanderse — APPROVED** [review]
Module looks great! Awesome work. Thanks for the effort you put in. 🎉

File: nixos/modules/services/web-apps/vikunja.nix (hunk 17 above)
Comment (em0lar) [reply nixos/modules/services/web-apps/vikunja.nix]: Okay, thanks for the hint, I removed the option

**Comment by em0lar** [comment]
* rebased on master

**Comment by em0lar** [comment]
@SuperSandro2000 Could you maybe take a look at it?

File: pkgs/servers/web-apps/vikunja/api.nix [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+      let
+        fakeGit = writeShellScriptBin "git" ''
+          if [[ $@ = "describe --tags --always --abbrev=10" ]]; then
+              echo "${version}"
+          else
+              >&2 echo "Unknown command: $@"
+              exit 1
+          fi
+        '';
+      in [ fakeGit mage ];
```
Comment (SuperSandro2000) [inline pkgs/servers/web-apps/vikunja/api.nix:25]: Could we patch this easier?

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by SuperSandro2000** [comment]
Not feeling comfortable to review if this is working correct or not.

File: pkgs/servers/web-apps/vikunja/api.nix (hunk 18 above)
Comment (em0lar) [reply pkgs/servers/web-apps/vikunja/api.nix:25]: It would be possible to always output `version`, but that would be kind of a dirty hack in my opinion. At least I don't know a easier way

**Comment by mkg20001** [comment]
I deployed this on my server and it works.

**Comment by kolaente** [comment]
🥳
